lib/src/source/git.dart

```diff
@@ -40,6 +41,49 @@ class GitSource extends CachedSource {
     });
   }
 
+  /// Gets the list of all versions from git tags.
+  /// Anything that Version.parse understands is considered a version,
+  /// it will also attempt to strip off a preceding 'v' e.g. v1.0.0
+  Future<List<Pubspec>> getVersions(String name, description) async {
+    if (!_useVersionTags(description)) {
+      // No need to get versions if the useVersionTags option is false
+      return super.getVersions(name, description);
+    }
+
+    PackageId id =
+        new PackageId(name, this.name, null, _getDescription(description));
+    String cachePath = _repoCachePath(id);
+    if (!entryExists(cachePath)) {
+      // Must have the repo cloned in order to list its tags
+      await _clone(_getUrl(id), cachePath);
+    }
+
+    List results = await git.run(["tag", "-l"], workingDir: cachePath);
+    List<Pubspec> validVersions = [];
+    for (String version in results) {
+      // Strip preceding 'v' character so 'v1.0.0' can be parsed into a Version
+      if (version.startsWith('v')) {
+        version = version.substring(1);
```

Comment:
This should do some de-duplication to ensure that, if a version exists both with and without `v`, only the canonical one (that is, the one without `v`) is returned.